### PR TITLE
Ks 2022 11 moderation tasks filter

### PR DESCRIPTION
**Pull request by Rineee**
@fuzzylogic2000 I already pushed this in case you need to go on with it..

The completed tasks can be added in the proposals admin. Although one can choose from any moderation task (not only the ones for the module), same for labels and categories, we should probably add a custom admin there.

I did not handle yet that only mod/initiators can view the filter.

### CODE REVIEW

**Comment by fuzzylogic2000** [comment]
> @fuzzylogic2000 I already pushed this in case you need to go on with it..
> 
> The completed tasks can be added in the proposals admin. Although one can choose from any moderation task (not only the ones for the module), same for labels and categories, we should probably add a custom admin there.
> 
> I did not handle yet that only mod/initiators can view the filter.

Yes, thank you! I will rebase on this branch. :)

File: meinberlin/apps/budgeting/api.py [hunk 1]
```diff
@@ -75,6 +77,14 @@ def list(self, request, *args, **kwargs):
                 'choices': label_choices,
             }
 
+        # moderation task filter
+        moderation_task_choices = self.get_moderation_task_choices()
```
Comment (Rineee) [inline meinberlin/apps/budgeting/api.py:81]: I first tried with adding a permission to the api, but that gets more complicated in frontend, so I thought it is easiest to simply only add the filter for moderators?

File: meinberlin/apps/budgeting/api.py [hunk 2]
```diff
@@ -75,6 +78,15 @@ def list(self, request, *args, **kwargs):
                 'choices': label_choices,
             }
 
+        # moderation task filter, only show to moderators
+        if is_allowed_moderate_project(request.user, self.module):
```
Comment (Rineee) [inline meinberlin/apps/budgeting/api.py:82]: I first tried with adding a permission to the api, but that gets more complicated in frontend, so I thought it is easiest to simply only add the filter for moderators?

Comment (fuzzylogic2000) [reply meinberlin/apps/budgeting/api.py:82]: Oh yes, I think that's much more elegant than only showing it to moderators in the frontend. :)

File: meinberlin/apps/ideas/admin.py [hunk 3]
```diff
@@ -18,5 +23,25 @@ class IdeaAdmin(module_admin.ItemAdmin):
     def has_add_permission(self, request, obj=None):
         return False
 
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
```
Comment (fuzzylogic2000) [inline meinberlin/apps/ideas/admin.py:26]: supercool! The whole admin thing!

**Review by fuzzylogic2000 — APPROVED** [review]
Supercool! 
Just one tiny thing: Can we move the task filter to the end of the filters? Or was it in the design like that?

**Comment by Rineee** [comment]
> Supercool! Just one tiny thing: Can we move the task filter to the end of the filters? Or was it in the design like that?

No, its at the end in the design! In my head it was somehow connected to categories and labels, but thats a total dev perspective :sweat_smile: Changed it now!

**Review by fuzzylogic2000 — APPROVED** [review]
Perfect!

**Pull request merged**